services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/director_v2/_thin_client.py

```diff
@@ -108,3 +110,15 @@ async def _(
             )
 
         return await _(self)
+
+    @retry_on_errors()
+    @expect_status(status.HTTP_200_OK)
+    async def get_dynamic_services(
+        self, *, user_id: UserID | None = None, project_id: ProjectID | None = None
```

Comment:
TIP: consider using this technique with `unset` sentinels and the helper function to build `params`
https://github.com/itisfoundation/osparc-simcore/blob/59aeb9964941234c040ac1c39eec9a8217db244d/services/web/server/src/simcore_service_webserver/folders/_folders_db.py#L314-L333

NOTE: if you use it , please move it from `utils` to `common_library`